// @mantine/core/cjs/components/Checkbox/CheckboxGroup/CheckboxGroup.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var hooks = require('@mantine/hooks');
var styles = require('@mantine/styles');
var InputWrapper = require('../../InputWrapper/InputWrapper.js');
var Checkbox = require('../Checkbox.js');
var Group = require('../../Group/Group.js');
var filterChildrenByType = require('../../../utils/filter-children-by-type/filter-children-by-type.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const defaultProps = {
  orientation: "horizontal",
  spacing: "lg",
  size: "sm"
};
const CheckboxGroup = React.forwardRef((props, ref) => {
  const _a = styles.useMantineDefaultProps("CheckboxGroup", defaultProps, props), {
    children,
    value,
    defaultValue,
    onChange,
    orientation,
    spacing,
    color,
    size,
    classNames,
    styles: styles$1,
    wrapperProps,
    errorProps,
    labelProps,
    descriptionProps
  } = _a, others = __objRest(_a, [
    "children",
    "value",
    "defaultValue",
    "onChange",
    "orientation",
    "spacing",
    "color",
    "size",
    "classNames",
    "styles",
    "wrapperProps",
    "errorProps",
    "labelProps",
    "descriptionProps"
  ]);
  const [_value, setValue] = hooks.useUncontrolled({
    value,
    defaultValue,
    finalValue: [],
    onChange,
    rule: (val) => Array.isArray(val)
  });
  const checkboxes = filterChildrenByType.filterChildrenByType(children, Checkbox.Checkbox).map((checkbox, index) => React.cloneElement(checkbox, {
    key: index,
    checked: _value.includes(checkbox.props.value),
    color,
    size,
    classNames,
    styles: styles$1,
    __staticSelector: "CheckboxGroup",
    onChange: (event) => {
      const itemValue = event.currentTarget.value;
      setValue(_value.includes(itemValue) ? _value.filter((item) => item !== itemValue) : [..._value, itemValue]);
    }
  }));
  return /* @__PURE__ */ React__default.createElement(InputWrapper.InputWrapper, __spreadValues(__spreadValues({
    labelElement: "div",
    size,
    __staticSelector: "CheckboxGroup",
    classNames,
    styles: styles$1,
    ref,
    errorProps,
    descriptionProps,
    labelProps
  }, wrapperProps), others), /* @__PURE__ */ React__default.createElement(Group.Group, {
    spacing,
    direction: orientation === "horizontal" ? "row" : "column",
    sx: { paddingTop: 5 }
  }, checkboxes));
});
CheckboxGroup.displayName = "@mantine/core/CheckboxGroup";

exports.CheckboxGroup = CheckboxGroup;
