// @mantine/core/esm/components/Checkbox/CheckboxGroup/CheckboxGroup.js
import React, { forwardRef, cloneElement } from 'react';
import { useUncontrolled } from '@mantine/hooks';
import { useMantineDefaultProps } from '@mantine/styles';
import { InputWrapper } from '../../InputWrapper/InputWrapper.js';
import { Checkbox } from '../Checkbox.js';
import { Group } from '../../Group/Group.js';
import { filterChildrenByType } from '../../../utils/filter-children-by-type/filter-children-by-type.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const defaultProps = {
  orientation: "horizontal",
  spacing: "lg",
  size: "sm"
};
const CheckboxGroup = forwardRef((props, ref) => {
  const _a = useMantineDefaultProps("CheckboxGroup", defaultProps, props), {
    children,
    value,
    defaultValue,
    onChange,
    orientation,
    spacing,
    color,
    size,
    classNames,
    styles,
    wrapperProps,
    errorProps,
    labelProps,
    descriptionProps
  } = _a, others = __objRest(_a, [
    "children",
    "value",
    "defaultValue",
    "onChange",
    "orientation",
    "spacing",
    "color",
    "size",
    "classNames",
    "styles",
    "wrapperProps",
    "errorProps",
    "labelProps",
    "descriptionProps"
  ]);
  const [_value, setValue] = useUncontrolled({
    value,
    defaultValue,
    finalValue: [],
    onChange,
    rule: (val) => Array.isArray(val)
  });
  const checkboxes = filterChildrenByType(children, Checkbox).map((checkbox, index) => cloneElement(checkbox, {
    key: index,
    checked: _value.includes(checkbox.props.value),
    color,
    size,
    classNames,
    styles,
    __staticSelector: "CheckboxGroup",
    onChange: (event) => {
      const itemValue = event.currentTarget.value;
      setValue(_value.includes(itemValue) ? _value.filter((item) => item !== itemValue) : [..._value, itemValue]);
    }
  }));
  return /* @__PURE__ */ React.createElement(InputWrapper, __spreadValues(__spreadValues({
    labelElement: "div",
    size,
    __staticSelector: "CheckboxGroup",
    classNames,
    styles,
    ref,
    errorProps,
    descriptionProps,
    labelProps
  }, wrapperProps), others), /* @__PURE__ */ React.createElement(Group, {
    spacing,
    direction: orientation === "horizontal" ? "row" : "column",
    sx: { paddingTop: 5 }
  }, checkboxes));
});
CheckboxGroup.displayName = "@mantine/core/CheckboxGroup";

export { CheckboxGroup };
